### ISSUE TITLE
Class with multiple lines of Generics

### ISSUE THREAD

**Issue by Rokt33r** [palantir/tslint#2952]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.4.0
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import {
  Action,
  Reducer,
} from './types'

export class MappedReducer<
  STATE,
  ACTION_TYPE = any,
  ACTION extends Action = Action
> {
  // skipped
}
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:recommended"
  ],"rules": {
    "indent": [true, "spaces", 2],
    "semicolon": [true, "never"],
    "quotemark": [true, "single"],
    "object-literal-sort-keys": false,
    "max-line-length": false,
    "interface-name": false,
    "no-namespace": false,
    "curly": [true, "ignore-same-line"],
    "array-type": false,
    "max-classes-per-file": false,
    "trailing-comma": [
      true,
      {
        "multiline": {
          "objects": "always",
          "arrays": "always",
          "functions": "always",
          "typeLiterals": "never"
        }
      }
    ]
  }
}
```

#### Actual behavior

`ERROR: ~/mapped-reducer.ts[10, 3]: misplaced opening brace`

#### Expected behavior

No error.

It doesn't throw any error if I write them as a single line, `export class MappedReducer<STATE, ACTION_TYPE = any, ACTION extends Action = Action> {}`.
